core/src/main/scala/mongo4cats/helpers.scala

```diff
@@ -38,62 +33,82 @@ private[mongo4cats] object helpers {
     def asyncSingle[F[_]: Async]: F[T] =
       Async[F].async_ { k =>
```

Comment:
We should be using `async` and cancel the subscription if the effect is canceled.

Another alternative might be to just use `boundedStream(1).take(1)`.